Entities/MovableObject.cpp

```diff
@@ -449,113 +449,254 @@ int MovableObject::Save(Writer &writer) const
 
-    // Clear out the instance object name so it gets created in the state upon first UpdateScript
-    m_ScriptObjectName.clear();
+        if (g_LuaMan.RunScriptString(m_ScriptPresetName + " = {};") < 0) {
+            return -3;
+        }
+
+        m_ScriptObjectName.clear();
+    }
 
-    // Under the class' table, create a new table for all functions of this specific preset and its unique ID
-    if ((error = g_LuaMan.RunScriptString(m_ScriptPresetName + " = {};")) < 0)
-        return error;
+    // Assign the different functions read in from the script to their permanent locations in the preset's table
+    for (std::string functionName : GetSupportedScriptFunctionNames()) {
+        if (g_LuaMan.GlobalIsDefined(functionName)) {
+            int error = g_LuaMan.RunScriptString(
+                m_ScriptPresetName + "." + functionName + " = " + m_ScriptPresetName + "." + functionName + " or {}; " +
+                m_ScriptPresetName + "." + functionName + "[\"" + scriptPath + "\"] = " + functionName + ";"
+            );
+
+            if (error < 0) {
+                return -3;
+            }
+        }
+    }
+    return 0;
+}
 
-    // Now assign the different functions read in from the script file to the permanent locations of this original preset's table
-    if ((error = g_LuaMan.RunScriptString("if Create then " + m_ScriptPresetName + ".Create = Create; end;")) < 0)
-        return error;
-    if ((error = g_LuaMan.RunScriptString("if Destroy then " + m_ScriptPresetName + ".Destroy = Destroy; end;")) < 0)
-        return error;
-    if ((error = g_LuaMan.RunScriptString("if Update then " + m_ScriptPresetName + ".Update = Update; end;")) < 0)
-        return error;
-	if ((error = g_LuaMan.RunScriptString("if OnPieMenu then " + m_ScriptPresetName + ".OnPieMenu = OnPieMenu; end;")) < 0)
-		return error;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+int MovableObject::ReloadScripts() {
+    if (m_LoadedScripts.empty()) {
+        return 0;
+    }
+
+    auto clearScriptConfigurationAndLoadPreexistingScripts = [](MovableObject *object) {
+        std::vector<std::pair<std::string, bool>> loadedScriptsCopy = object->m_LoadedScripts;
+        object->m_LoadedScripts.clear();
+        object->m_ScriptPresetName.clear();
+
+        int status = 0;
+        for (std::pair<std::string, bool> scriptEntry : loadedScriptsCopy) {
+            status = object->LoadScript(scriptEntry.first, scriptEntry.second);
+            if (status < 0) {
+                return status;
+            }
+        }
+        return status;
+    };
+
+    int status = clearScriptConfigurationAndLoadPreexistingScripts(this);
+    if (status >= 0) {
+        return status;
+    }
+    MovableObject *pPreset = const_cast<MovableObject *>(dynamic_cast<const MovableObject *>(g_PresetMan.GetEntityPreset(GetClassName(), GetPresetName(), GetModuleID())));
```

Comment:
Oof this just looks bad.

Reply:
Not my fault! This came from data and I just left it. As you say, It's ugly as sin and it makes sonarlint upset. I guess the way to solve it would be add a non const GetEntityPreset, or do some funny back and forth through a new method in PresetMan. The options should probably be investigated first I guess.

Reply:
¯\\\_(ツ)_/¯

Reply:
I'll put a TODO in